aten/src/ATen/core/type.cpp

```diff
@@ -11,6 +11,26 @@
 #include <ATen/core/function.h>
 #include <iostream>
 
+namespace std {
+template<>
+struct hash<std::tuple<std::string, c10::TypePtr, c10::TypePtr>> {
+  size_t operator()(std::tuple<std::string, c10::TypePtr, c10::TypePtr> const& t) const {
+    // This hashing is all hidden behind a static initializer so it
+    // doesn't have to be optimal
```

Comment:
great, might be a good idea to use hash_combine then

Reply:
Just saw that we have an `at::hash_combine` - I'll use that, thanks!